server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1008,6 +1013,14 @@ protected Node(
                 searchModule.getIndexSearcherExecutor(threadPool)
             );
 
+            if (FeatureFlags.isEnabled(TRACER)) {
+                final TracerSettings tracerSettings = new TracerSettings(settings, clusterService.getClusterSettings());
+                List<TracerPlugin> tracerPlugins = pluginsService.filterPlugins(TracerPlugin.class);
+                TracerModule tracerModule = new TracerModule(settings, tracerPlugins, threadPool, tracerSettings);
+                TracerManager.initTracerManager(tracerSettings, tracerModule.getTracerSupplier(), tracerModule.getTracerHeaderInjector());
```

Comment:
Please inject this(Binder#bind) rather than creating a static singleton instance

Reply:
Agree to your uber point to create beans/objects through Binder. But in our use-case we need to have a handle of instance of TracerManager to be used by users to create/ends spans. Hence we chose this approach. This is how users will be using our code

```
TracerManager.getTracer().startSpan("span name", Level.Info);
...
TracerManager.getTracer().endSpan();

```

If we go with former approach, we will have to inject the TracerManager instance in each class where we want to create spans.

Reply:
Addressed the comments